### ISSUE TITLE
Pylint Fails to output W0199 on single item tuple

### ISSUE THREAD

**Issue by JasonF-BJSS** [PyCQA/pylint#4655]
### Steps to reproduce

Given a file `a.py`:
```python
# pylint: disable=missing-docstring
assert (1 == 2, "r")
assert (1 == 2,)
```

### Current behaviour

Result of `pylint a.py`:
```
************* Module a
a.py:2:0: W0199: Assert called on a 2-item-tuple. Did you mean 'assert x,y'? (assert-on-tuple)
```

### Expected behavior

```
************* Module a
a.py:2:0: W0199: Assert called on a 2-item-tuple. Did you mean 'assert x,y'? (assert-on-tuple)
a.py:3:0: W0199: Assert called on a 2-item-tuple. Did you mean 'assert x,y'? (assert-on-tuple)
```

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.6.2
astroid 2.4.2
Python 3.8.7 (v3.8.7:6503f05dd5, Dec 21 2020, 12:45:15) 
[Clang 6.0 (clang-600.0.57)]
```

**Comment by yushao2** [comment]
@cdce8p, this seems to be expected behavior from the already implemented functional test:

https://github.com/PyCQA/pylint/blob/main/tests/functional/a/assert_on_tuple.py#L8

**Comment by cdce8p** [comment]
> @cdce8p, this seems to be expected behavior from the already implemented functional test:

Indeed, thanks for the lookup! I didn't look at it closely when adding the tag.
Just the question if it should change then otherwise this issue could be closed.

**Comment by Pierre-Sassoulas** [comment]
What is the legitimate use case of ``assert (1 == 2,)`` ?

**Comment by JasonF-BJSS** [comment]
There is no legitimate use case I could ever see for` assert (1 == 2,)`, the linter should warn as per `assert (1 == 2, "r")`.

**Comment by Pierre-Sassoulas** [comment]
It's minor as like @JasonF-BJSS said, python itself is raising a warning, so it will be caught when the code is run but pylint needs to warn beforehand. So the functional tests are wrong, we need to remove them and raise a warning.
